flink-core/src/main/java/org/apache/flink/metrics/reporter/JMXReporter.java

```diff
@@ -265,4 +329,72 @@ public Object getValue() {
 			return gauge.getValue();
 		}
 	}
+
+	/**
+	 * JMX Server implementation that JMX clients can connect to.
+	 *
+	 * Heavily based on j256 simplejmx project
+	 *
+	 * https://github.com/j256/simplejmx/blob/master/src/main/java/com/j256/simplejmx/server/JmxServer.java
+	 */
+	private static class JMXServer {
+		private int port;
+		private Registry rmiRegistry;
```

Comment:
Hmm, if we don't understand the code and, thus, cannot explain what it does, we shouldn't include the code. The question who should maintain this kind of code. I think it would be best if you could do some research to clarify how the `JMXServer` works and add some comments explaining the different components.